curl/tests/client_test.cc

```diff
@@ -469,15 +473,15 @@ namespace leatherman { namespace curl {
              test_impl->test_failure_mode = curl_impl::error_mode::set_url_error;
              REQUIRE_THROWS_AS(test_client.download_file(req, file_path), curl::http_file_download_exception);
```

Comment:
Sure. I'd rather this were a flatter taxonomy, and since no one's likely using this I'd be ok breaking it a bit.

I'd see something like
http_exception
- http_request_exception
- http_setup_exception
- http_file_exception